app/controllers/spree/admin/general_settings_controller.rb

```diff
@@ -19,6 +20,18 @@ def update
 
         redirect_to edit_admin_general_settings_path
       end
+
+      private
+
+      def combine_available_units_params
+        available_units = []
+        params.each do |name, value|
+          if value == "1" && unit = name.match(/available_units_(.*)/)&.captures&.first
```

Comment:
Is there a way we can do this without using regexps? Can we use nested params? Something like params[:available_units][:g] ?

If we end up with this code, I think this if clause is too complicated and needs to be extracted to a method with a good name. Maybe:

```ruby
def extract_unit_from_param_name(param_name)
   param_name.match(/available_units_(.*)/)&.captures&.first
end
```
and then:

```ruby
if value == "1" && unit = extract_unit_from_param_name(name)
```

Now I see it's an assignment in an if clause, I thought it was a ==. Is rubocop not complaining about this?
In this case I recommend:
```ruby
if value == "1"
   unit = extract_unit_from_param_name(name)
   available_units << unit if unit
end
```

Reply:
Good thinking - I didn't think to do nested params since there's not an ActiveRecord model that's backing this up, and there's not an option for fields_for in the base_helper.rb.

But! We can have it *act* like nested params and avoid the regex for sure: a0d4082d01dded8b81d2224c39e81a5e6c1cba99

I left it as a separated method (renamed to `merge_available_units_params` even though it's now just a one-liner.